Library/Homebrew/cmd/audit.rb

```diff
@@ -225,9 +246,35 @@ def audit_urls
       problem "\"ftpmirror.gnu.org\" is preferred for GNU software (url is #{u})."
     end
 
-    # the rest of the checks apply to mirrors as well
+    # the rest of the checks apply to mirrors as well.
     urls.concat(@specs.map(&:mirrors).flatten)
 
+    # Check a variety of SSL/TLS links that don't consistently auto-redirect.
+    urls.each do |p|
+      # Skip the main url link, as it can't be made SSL/TLS yet.
+      next if p =~ %r[/ftpmirror\.gnu\.org]
+
+      # No auto-direct as of Jan 2015.
+      if p =~ %r[^http://ftp\.gnu\.org]
+        problem "ftp.gnu.org links should be https://, not http:// (url is #{p})."
+      end
+
+      # This does redirect, but it's an incredibly common linking mistake.
+      if p =~ %r[^http://code\.google\.com]
+        problem "code.google.com links should be https://, not http (url is #{p})."
+      end
+
+      # No auto-direct as of Jan 2015.
+      if p =~ %r[^http://fossies\.org]
+        problem "Fossies links should be https://, not http (url is #{p})."
+      end
+
+      # No auto-direct as of Jan 2015.
+      if p =~ %r[^http://mirrors\.kernel\.org/]
+        problem "mirrors.kernel links should be https://, not http (url is #{p})."
+      end
```

Comment:
Neat. Should help make the block look a little tighter. Thanks!

Sent from my iPhone

> On 5 Jan 2015, at 04:43, Jack Nagel <notifications@github.com> wrote:
> 
> In Library/Homebrew/cmd/audit.rb:
> 
> > +      end
> > +
> > +      # This does redirect, but it's an incredibly common linking mistake.
> > +      if p =~ %r[^http://code\.google\.com]
> > +        problem "code.google.com links should be https://, not http (url is #{p})."
> > +      end
> > +
> > +      # No auto-direct as of Jan 2015.
> > +      if p =~ %r[^http://fossies\.org]
> > +        problem "Fossies links should be https://, not http (url is #{p})."
> > +      end
> > +
> > +      # No auto-direct as of Jan 2015.
> > +      if p =~ %r[^http://mirrors\.kernel\.org/]
> > +        problem "mirrors.kernel links should be https://, not http (url is #{p})."
> > +      end
> yep.
> 
> —
> Reply to this email directly or view it on GitHub.
>